src/catnap/futures/mod.rs

```diff
@@ -55,7 +52,7 @@ pub enum OperationResult {
     Connect,
     Accept(RawFd),
     Push,
-    Pop(Option<(Ipv4Addr, Port16)>, Bytes),
+    Pop(Option<Ipv4Endpoint>, Bytes),
```

Comment:
Glad you chose to standardize on the Endpoint option versus the Address, Port combo.

But looking to the future, would it be possible to standardize on a more generic IpEndpoint here, so we don't have to change this again to add IPv6 support?

Reply:
Yeah, it would be better to have it based in a generic `IpEndpoint` structure. Unfortunately, catnip was built on top of the `Ip´v4Endpoint` structure and we should first change this.

I've filed an issue for this: https://github.com/demikernel/catnip/issues/new

Reply:
Great, thanks.